src/main/webapp/src/app/schedule-page/datepicker/datepicker.component.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnInit} from '@angular/core';
+import {MatDatepickerInputEvent} from '@angular/material/datepicker';
+import {FormControl} from '@angular/forms';
+
+@Component({
+  selector: 'datepicker',
+  templateUrl: './datepicker.component.html',
+})
+export class DatepickerComponent implements OnInit {
+  date = new FormControl(new Date());
+  currDate: string = this.formatDate(this.date.value);
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  /* Updates the displayed date whenever user chooses a date in the datepicker */
+  updateCurrDate(event: MatDatepickerInputEvent<Date>): void {
+    if (event.value) {
+      this.currDate = this.formatDate(`${event.value}`);
+    } else {
+      this.currDate = '';
+    }
+  }
+
+  /**
+   * Formats a string containing a date to: {Day Abbreviation}: {Month} {Day}, {Year}
+   * @param {string} date - a string separated by spaces with the following info {Day} {Month} {Day} {Year} {Hour:Minute:Seconds} {GMT} {Timezone}
+   */
+  formatDate(date: string): string {
+    let formattedDate = '';
+    if (date) {
```

Comment:
do we need the if check? won't `date` always exist?

Reply:
The check was added in case the user decides to remove the date from the datepicker in which then the date becomes an empty string

Reply:
But aren't you handling that here as well? 

https://github.com/googleinterns/step26-2020/blob/3ea1cad296ea7f3c3ffd6c32eb6000a45d837e79/src/main/webapp/src/app/schedule-page/datepicker/datepicker.component.ts#L33-L34